src/pages/home/report/ReportActionsList.tsx

```diff
@@ -250,7 +250,9 @@ function ReportActionsList({
      */
     const wasMessageReceivedWhileOffline = useCallback(
         (message: OnyxTypes.ReportAction) =>
-            !wasActionTakenByCurrentUser(message) && wasActionCreatedWhileOffline(message, isOffline, lastOfflineAt.current, lastOnlineAt.current, preferredLocale),
+            !wasActionTakenByCurrentUser(message) &&
+            wasActionCreatedWhileOffline(message, isOffline, lastOfflineAt.current, lastOnlineAt.current, preferredLocale) &&
+            !(message.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD || message.isOptimisticAction),
```

Comment:
I proceed with a different approach because I feel like having the `isOffline` condition in `wasActionCreatedWhileOffline` makes sense with the function name. So, I updated `wasMessageReceivedWhileOffline` instead because if it's an optimistic action, then it's a message that we send, not receive.